lib/widgets/lightbox.dart

```diff
@@ -197,22 +199,207 @@ class _LightboxPageState extends State<_LightboxPage> {
+    // won't change while user is on the lightbox page but we still
+    // handle the reinitialization for correctness.
+    if (_controller != null) _deinitialize();
+    _initialize();
+  }
+
+  Future<void> _initialize() async {
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    assert(debugLog('VideoPlayerController.networkUrl(${widget.src})'));
+    _controller = VideoPlayerController.networkUrl(widget.src, httpHeaders: {
+      if (widget.src.origin == store.account.realmUrl.origin) ...authHeader(
+        email: store.account.email,
+        apiKey: store.account.apiKey,
+      ),
+      ...userAgentHeader()
+    });
+    _controller!.addListener(_handleVideoControllerUpdates);
+
+    try {
+      await _controller!.initialize();
+      await _controller!.play();
+    } catch (error) { // TODO(log)
+      assert(debugLog("VideoPlayerController.initialize failed: $error"));
+      if (mounted) {
+        await showErrorDialog(
+          context: context,
+          title: zulipLocalizations.errorDialogTitle,
+          message: zulipLocalizations.errorVideoPlayerFailed,
+          // To avoid showing the disabled video lightbox for the unnsupported
+          // video, we make sure user doesn't reach there by dismissing the dialog
+          // by clicking around it, user must press the 'OK' button, which will
+          // take user back to content message list.
+          barrierDismissible: false,
+          onContinue: () {
+            Navigator.pop(context); // Pops the dialog
+            Navigator.pop(context); // Pops the lightbox
+          });
+      }
+    }
+  }
+
+  @override
+  void dispose() {
+    _deinitialize();
+    super.dispose();
+  }
+
+  void _deinitialize() {
+    _controller?.removeListener(_handleVideoControllerUpdates);
+    _controller?.dispose();
+    _controller = null;
+  }
+
+  void _handleVideoControllerUpdates() {
+    setState(() {});
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _LightboxPageLayout(
+      routeEntranceAnimation: widget.routeEntranceAnimation,
+      message: widget.message,
+      buildBottomAppBar: (context, color, elevation) =>
+        _controller == null
+          ? null
+          : BottomAppBar(
+              height: 150,
+              color: color,
+              elevation: elevation,
+              child: Column(mainAxisAlignment: MainAxisAlignment.end, children: [
+              Row(children: [
+                Text(_formatDuration(_controller!.value.position),
+                  style: const TextStyle(color: Colors.white)),
+                Expanded(child: Slider(
+                  value: _controller!.value.position.inSeconds.toDouble(),
+                  max: _controller!.value.duration.inSeconds.toDouble(),
+                  activeColor: Colors.white,
+                  onChanged: (value) {
+                    _controller!.seekTo(Duration(seconds: value.toInt()));
```

Comment:
Interesting, this puts everything into whole seconds. What's the reasoning for that?

This behavior seems probably fine, except for one case where it seems like it could be annoying: if you try to seek to the end using the slider, it'll be a little bit before the end. If the video happens to be 2.9 seconds long, or 1234.9 seconds, then the slider will only get you to 900ms away from the end, no matter how hard you drag it to the end.

What happens if we just use the times with as much precision as we get them with?

Reply:
Initial reasoning was that interactivity of seeking would be much laggy if we did `seekTo` on every 'onChange' for higher precisions.

Now the slider uses millisecond precision, which seems to be what we are getting from the controller (by logging the controller.position). Also the call to `controller.seekTo` now happens in `onChangeEnd` which now seems like what should have been done initially anyway.